File: fe/mobile/android/app/src/main/java/com/capstone/data/repository/MediaRepositoryImpl.java
package com.capstone.data.repository;

import com.capstone.data.source.remote.GoogleCloudUploader;
import com.capstone.data.source.remote.MediaApi;
import com.capstone.data.source.request.SaveImageUrlRequest;
import com.capstone.domain.repository.MediaRepository;

import java.io.File;

public class MediaRepositoryImpl implements MediaRepository {
    private final MediaApi mediaApi;
    private final GoogleCloudUploader gcsUploader;

    public MediaRepositoryImpl(MediaApi mediaApi, GoogleCloudUploader gcsUploader) {
        this.mediaApi = mediaApi;
        this.gcsUploader = gcsUploader;
    }

    @Override
    public String processCapturedImage(File file) throws Exception {
        // 1. Tải ảnh lên Google Cloud Storage trước
        var gcsUrl = gcsUploader.uploadImage(file);

        // 2. Lưu URL của ảnh vào hệ thống Backend
        mediaApi.saveImageUrl(new SaveImageUrlRequest(gcsUrl, null)).execute();

        return gcsUrl;
    }

    @Override
    public String performOcr(String imageUrl) throws Exception {
        // Placeholder cho chức năng OCR trong tương lai
        return "";
    }
}

File: fe/mobile/android/app/src/main/java/com/capstone/data/repository/MediaRepositoryImpl.kt
package com.capstone.data.repository

import com.capstone.data.source.remote.GoogleCloudUploader
import com.capstone.data.source.remote.MediaApi
import com.capstone.data.source.request.SaveImageUrlRequest
import com.capstone.domain.repository.MediaRepository
import java.io.File

class MediaRepositoryImpl(
    private val mediaApi: MediaApi,
    private val gcsUploader: GoogleCloudUploader
) : MediaRepository {

    /**
     * Quy trình xử lý ảnh: Upload lên Google Cloud trước, sau đó lưu URL nhận được vào Backend.
     */
    override suspend fun processCapturedImage(file: File): String {
        val gcsUrl = gcsUploader.uploadImage(file)
        mediaApi.saveImageUrl(SaveImageUrlRequest(imageUrl = gcsUrl))
        return gcsUrl
    }

    /**
     * Placeholder cho chức năng OCR trong tương lai.
     */
    override suspend fun performOcr(imageUrl: String): String {
        return ""
    }
}
